internal/fs/walk.go

```diff
@@ -41,26 +45,44 @@ func Walk(path string, opts *WalkOptions) (*types.Node, error) {
 			log.Warningf("%s: %v", path, err)
 			return root, nil
 		}
+
+		eg, _ := errgroup.WithContext(context.Background())
+		rl := &sync.Mutex{}
+
 		for _, name := range names {
-			child, err := Walk(path+"/"+name, opts)
-			if err != nil {
-				return nil, err
-			}
-			child.Parent = root
-			root.Size += child.Size
-			root.SizeOnDisk += child.SizeOnDisk
-			threshOK := child.Size >= opts.SizeThreshold
-			ignoreOK := true
-			if opts.IgnoreChecker != nil && opts.IgnoreChecker.ShouldIgnore(child.Name, child.IsDir) {
-				ignoreOK = false
-			}
-			if !ignoreOK {
-				log.Debugf("ignore: %s", path+"/"+child.Name)
-			}
-			if threshOK && ignoreOK {
-				root.Children = append(root.Children, child)
-			}
+			var curr = name
+			eg.Go(func() error {
+				child, err := Walk(path+"/"+curr, opts)
+				if err != nil {
+					return err
+				}
+				child.Parent = root
+				rl.Lock()
```

Comment:
instead of unlocking at the end of the function, let's use `defer` right after locking

```suggestion
				rl.Lock()
				defer rl.Unlock()
```